fling/model/cnn.py

```diff
@@ -32,10 +33,26 @@ def __init__(
 
         self.glp = nn.AdaptiveAvgPool2d((1, 1))
 
-        self.fc = nn.Sequential(nn.Flatten(), nn.Linear(hidden_dims[-1], class_number))
-
-    def forward(self, x):
+        if len(linear_hidden_dims):
+            self.mlp = []
+            self.mlp.append(nn.Flatten())
+            self.mlp.append(nn.Linear(hidden_dims[-1], linear_hidden_dims[0]))
+            for i in range(len(linear_hidden_dims) - 1):
+                self.mlp.append(get_activation(name=activation))
+                self.mlp.append(nn.Linear(linear_hidden_dims[i], linear_hidden_dims[i + 1]))
+            self.mlp = nn.Sequential(*self.mlp)
+            self.fc = nn.Linear(linear_hidden_dims[-1], class_number)
+        else:
+            self.mlp = None
+            self.fc = nn.Sequential(nn.Flatten(), nn.Linear(hidden_dims[-1], class_number))
+
+    def forward(self, x, mode='compute-logit'):
         x = self.layers(x)
         x = self.glp(x)
-        x = self.fc(x)
-        return x
+        if self.mlp:
+            x = self.mlp(x)
+        y = self.fc(x)
+        if mode == 'compute-feature-logit':
+            return x, y
+        else:
```

Comment:
List all possible cases like 'compute-feature-logit' or 'compute-logit'. Add an exception if no legal mode is provided.